main.cpp

```diff
@@ -495,17 +725,33 @@ struct Sampler
     Envelope ampEnv;
     LFO lfo01;
     Waveform waveform;
+    float loopCounter = 0.f;
 
     void loopSound(std::string sampleName, bool loopForward);
     void modulateFilter(Filter filter, LFO lfo);
     void pitchDown(std::string sampleName);
+    float countLoops(int msPlayingTime = 10);
 };
 
 Sampler::Sampler()
 {
     std::cout << "Sampler constructed." << "\n";
 }
 
+float Sampler::countLoops(int msPlayingTime)
+{
+    waveform.samplesMs = waveform.lengthInSamples / waveform.sampleRate;
+
+    for (float i = 0.f; i <= float(msPlayingTime); ++i)
+    {
+        std::cout << "LOOP COUNTER -> Time passed: " << i << "ms | Loops passed (floored): " << floor(double(i) / double(waveform.samplesMs)) << "\n";
+    }
+
+    std::cout << "If you play the sample for " << msPlayingTime << "ms. It'll loop " << float(msPlayingTime) / waveform.samplesMs << " times\n";
+
+    return float(msPlayingTime) / waveform.samplesMs;
+}
```

Comment:
Casting has not been introduced yet, but I'll allow this because nothing else needs fixing, and because these are functional casts, and not C-style casts.